lib/active_shipping/tracking_response.rb

```diff
@@ -101,5 +101,21 @@ def has_exception?
     alias_method :scheduled_delivery_time, :scheduled_delivery_date
     alias_method :actual_delivery_time, :actual_delivery_date
     alias_method :attempted_delivery_time, :attempted_delivery_date
+
+    def ==(other)
+      carrier == other.carrier && carrier_name == other.carrier_name && status == other.status &&
```

Comment:
See the above comment. This would be even better for the iteration, to be honest:

``` ruby
attributes = [:carrier,:carrier_name, :status,:status_code, :status_description, :ship_time, :scheduled_delivery_date, :actual_delivery_date, :attempted_delivery_date, :delivery_signature, :tracking_number, :shipment_events, :shipper_address, :origin, :destination]
attributes.all? { |attr| self.public_send(:attr) == other.public_send(:attr) }
```